### PR TITLE
Fix Windows Compiling Issues

### PR DESCRIPTION
**Pull request by dabonnie**
Add Windows visibility
Fix Windows warnings

Signed-off-by: Devin Bonnie <dbbonnie@amazon.com>

### CODE REVIEW

File: libstatistics_collector/include/libstatistics_collector/topic_statistics_collector/constants.hpp [hunk 1]
```diff
@@ -17,6 +17,8 @@
 
 #include <string>
 
+#include "libstatistics_collector/visibility_control.hpp"
```
Comment (prajakta-gokhale) [inline libstatistics_collector/include/libstatistics_collector/topic_statistics_collector/constants.hpp:20]: Needed? Visibility macros don't seem to be used in this file?

Same comment for all files that include the header but don't use the macros.

File: libstatistics_collector/test/topic_statistics_collector/test_received_message_age.cpp [hunk 2]
```diff
@@ -97,8 +97,11 @@ TEST(ReceivedMessageAgeTest, TestAgeMeasurement) {
   EXPECT_TRUE(test_collector.IsStarted()) << "Expect to be started";
 
   rcl_time_point_value_t fake_now_nanos_{kStartTime};
+
   auto msg = DummyMessage{};
-  msg.header.stamp.nanosec = fake_now_nanos_;
+  msg.header.stamp.sec = static_cast<std::int32_t>(RCL_NS_TO_S(fake_now_nanos_));
```
Comment (prajakta-gokhale) [inline libstatistics_collector/test/topic_statistics_collector/test_received_message_age.cpp:102]: Is this field needed anywhere in the tests?

File: libstatistics_collector/include/libstatistics_collector/collector/collector.hpp [hunk 3]
```diff
@@ -19,6 +19,8 @@
 #include <string>
 
 #include "metric_details_interface.hpp"
+
+#include "libstatistics_collector/visibility_control.hpp"
```
Comment (prajakta-gokhale) [inline libstatistics_collector/include/libstatistics_collector/collector/collector.hpp:23]: Did you add this file in the PR?

File: libstatistics_collector/test/topic_statistics_collector/test_received_message_age.cpp (hunk 2 above)
Comment (dabonnie) [reply libstatistics_collector/test/topic_statistics_collector/test_received_message_age.cpp:102]: It's used in the `OnMessageReceived` where the `DummyMessage` nanoseconds is reconstructed from the message header.

File: libstatistics_collector/include/libstatistics_collector/topic_statistics_collector/constants.hpp (hunk 1 above)
Comment (dabonnie) [reply libstatistics_collector/include/libstatistics_collector/topic_statistics_collector/constants.hpp:20]: Will remove where unused.

File: libstatistics_collector/include/libstatistics_collector/collector/collector.hpp (hunk 3 above)
Comment (dabonnie) [reply libstatistics_collector/include/libstatistics_collector/collector/collector.hpp:23]: Should be added now!

File: libstatistics_collector/include/libstatistics_collector/topic_statistics_collector/received_message_period.hpp [hunk 4]
```diff
@@ -20,6 +20,7 @@
 #include <string>
 
 #include "constants.hpp"
+
```
Comment (prajakta-gokhale) [inline libstatistics_collector/include/libstatistics_collector/topic_statistics_collector/received_message_period.hpp:23]: Nit: these belong in the same group, no newline needed.

**Comment by dabonnie** [comment]
> Looks fine to me - but do we need dllexport on private members?

We don't just verified. Will remove.

**Pull request merged**